topdown/eval_test.go

```diff
@@ -804,121 +929,6 @@ func TestPartialRule(t *testing.T) {
 			query:  `data = x`,
 			expErr: "eval_conflict_error: object keys must be unique",
 		},
-		// NOTE: There is a bit of ambiguity in the parser about when an else-body is permitted.
-		// Sometimes when successfully parsed, evaluation doesn't consider the else body, though.
-		// else bodies
```

Comment:
Has this been resolved? 🤔

Reply:
From what I gather, `else` bodies aren't supported for partial rules. The confusion came from `else` being supported for ref-head rules that don't overlap with a partial rule.

Reply:
It could be that the parser/compiler is a bit too lenient, though, and could be tightened to reject additional invalid scenarios.